Automattic/simplenote-macos
# Breadcrumbs

### Description:
In this PR we're implementing a new StatusBar, in charge of rendering context information.

Closes #979 
cc @eshurakov

### Details:
I was in the neighborhood, and ended up dropping few Notification based callbacks. Explicit flows, channeled thru the AppDelegate, have been implemented.

### Test: Main UX
1. Launch the app
2. Click over any note

- [x] Verify that the StatusBar shows the current `Tag > Note Title`
- [x] Verify that whenever the Note Title is longer than 60 characters, the text gets clipped, and shows a trailing `...`
- [x] Verify that clicking over an empty Tag causes the StatusBar to only show the Tag Name
- [x] Verify the new StatusBar reacts well to Light / Dark Mode
- [x] Verify that the Notes List area does not clip the last TableViewCell

### Test: Highlight
1. Click over the Tags List

- [x] Verify that the StatusBar highlights the active **Tag** 
- [x] Verify that clicking over the Notes / Editor area causes the StatusBar to highlight the **Note Title** instead

### Test: Multiple Selection
1. Control click and select multiple Notes in the notes List

- [x] Verify that the StatusBar reads `Tag / X Selected`

### Test: Search
1. Enter a Search Keyword

- [x] Verify that the StatusBar shows `Searching "keyword"`
- [x] Verify that Search Results are presented in the Notes List area
- [x] Verify that the Editor also highlights relevant keywords
- [x] Verify that clicking over any other tag causes the Search Mode to be dismissed

### Release
`RELEASE-NOTES.txt` was updated in e9f7b2d with:

> New Status Bar: Lets you quickly know what's your Tags / Notes selection #979
